test/bin/missing-dune-project/run.t

```diff
@@ -0,0 +1,21 @@
+Let us setup a simple project with two local packages, defined at the root
```

Comment:
Since this test is self-contained (you create necessary files in the test itself) you can just name it `test/bin/missing-dune-project.t`.

Don't forget to move the dune file up!

Reply:
I'm not sure I'd want to mix files and directories here. Having a directory as the adavantage to be more future proof here as for more complex test cases, it might be good to have a directory inside which you CD and run the command yourself.

Reply:
I read the documentation for cram tests and I'm a bit surprised they went for this. It seems that if you don't use the `dir.t/` feature you'll have to declare the dependencies of your test by hand so the best solution, in terms of consistency, would be to put all tests in a `test-name.t/` folder, whether they have extra files or not.

I also wonder if it wouldn't make more sense to always put the files we need in the directory rather than generating them on the fly as part of the test.

Reply:
@Leonidas-from-XIV any thoughts on this?

Reply:
@NathanReb I think it would indeed be a bit nicer to have the files in there, because the tests often spend quite some (extremely repetitive) lines to create the content.

What I am less happy about is that files like `.gitignore` would be there too and picked up by the "outer" project git, thus effectively often ignoring `run.t` which we actually don't want to have ignored. As a compromise, the `.gitignore` file could be written/`mv`'d inside the test. I am not amazed by that idea, but maybe it is a viable 80% solution? What do you think?

Reply:
Yeah that sounds good!